cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/GoalViolationDetector.java

```diff
@@ -139,12 +143,19 @@ public void run() {
                                                              clusterModel.capacityEstimationInfoByBrokerId());
             _lastCheckedModelGeneration = clusterModel.generation();
           }
-          Set<Integer> excludedBrokersForLeadership = EXCLUDE_RECENTLY_DEMOTED_BROKERS
-                                                      ? _kafkaCruiseControl.state(new OperationProgress(), Collections.singleton(EXECUTOR))
-                                                                           .executorState().recentlyDemotedBrokers()
-                                                      : Collections.emptySet();
 
-          newModelNeeded = optimizeForGoal(clusterModel, goal, goalViolations, excludedBrokersForLeadership);
+          ExecutorState executorState = null;
```

Comment:
should we move this out of the for loop(calculate exclude brokers once and reuse by all self-healing goals)?